### PR TITLE
Api/post put

### PR DESCRIPTION
**Pull request by hewonjeong**
- API (Post / Put) function include tests

### CODE REVIEW

File: src/server/items/controllers.js [hunk 1]
```diff
@@ -93,5 +93,81 @@ export default {
       }
       return cb();
     });
+  },
+  add: (req, res, cb) => {
+    let increment;
```
Comment (ovekyc) [inline src/server/items/controllers.js]: Please move this variable declaration to 107line and make it const

File: src/server/items/controllers.js [hunk 2]
```diff
@@ -93,5 +93,81 @@ export default {
...
+    db.get('itemIncrement', (getErr, value) => {
+      if (getErr) {
+        res.status(500);
+        res.send({
+          error: getErr
+        });
+        cb();
+      } else {
+        increment = value + 1;
```
Comment (ovekyc) [inline src/server/items/controllers.js]: const

File: src/server/items/controllers.js [hunk 3]
```diff
@@ -93,5 +93,80 @@ export default {
...
+            cb();
+          } else {
+            db.put('item' + increment, req.body, (itemErr) => {
+              if (itemErr) {
+                res.status(500);
+                res.send({
+                  error: itemErr
+                });
+                cb();
+              } else {
```
Comment (fguy) [inline src/server/items/controllers.js]: There's way to reduce indentation level with 'return' statement.

File: src/server/database.js [hunk 4]
```diff
@@ -1,4 +1,13 @@
...
+const db = levelup(config.database, {valueEncoding: 'json'});
+
+db.get('itemIncrement', (err) => {
+  if (err && err.type === 'NotFoundError') {
+    db.put('itemIncrement', 0);
+  }
+});
+
+export default db;
+
```
Comment (fguy) [inline src/server/database.js]: nit; please remove the extra line.

File: src/server/items/controllers.js [hunk 5]
```diff
@@ -93,5 +93,80 @@ export default {
       }
       return cb();
     });
+  },
+  add: (req, res, cb) => {
+    db.get('itemIncrement', (getErr, value) => {
+      if (getErr) {
+        res.status(500);
```
Comment (fguy) [inline src/server/items/controllers.js]: You can chain status and send.

`res.status(500).send(...)`

File: src/server/items/controllers.js [hunk 6]
```diff
@@ -93,5 +93,80 @@ export default {
...
+        res.send({
+          error: getErr
+        });
+        cb();
+      } else {
+        const increment = value + 1;
```
Comment (fguy) [inline src/server/items/controllers.js]: Why do you need this incremental value here? https://www.quora.com/How-do-you-generate-monotonically-increasing-sequence-numbers-seqnums-in-a-large-distributed-system

File: src/server/database.js [hunk 7]
```diff
@@ -1,4 +1,6 @@
 import levelup from 'levelup';
 import config from 'config';
 
-export default levelup(config.database, {valueEncoding: 'json'});
+const db = levelup(config.database, {valueEncoding: 'json'});
+
+export default db;
```
Comment (ovekyc) [inline src/server/database.js]: you can export db just 'default levelup..'

File: src/server/items/controllers.js [hunk 8]
```diff
@@ -81,17 +64,45 @@ export default {
...
-        res.send({
-          message: 'success'
-        });
+        res.status(200).send({message: 'success'});
       }
       return cb();
     });
+  },
+  add: (req, res, cb) => {
+    const itemId = 'item-' + uuid();
```
Comment (ovekyc) [inline src/server/items/controllers.js]: please use template

Comment (hewonjeong) [reply src/server/items/controllers.js]: @ovekyc thx to comments. It will be modified

File: src/test/node/items/controllers.js [hunk 9]
```diff
@@ -234,5 +239,4 @@ test('delete all item from database', t => {
       });
     });
   });
-});
-
+});
```
Comment (hewonjeong) [inline src/test/node/items/controllers.js]: add new line

File: src/server/items/routes.js [hunk 10]
```diff
@@ -208,4 +208,4 @@ router.put('/:id', ItemContollers.modify);
  */
 router.delete('/:id', ItemContollers.remove);
 
-export default router;
+export default router;
```
Comment (hewonjeong) [inline src/server/items/routes.js]: add new line

**Comment by fguy** [comment]
LGTM

**Comment by ovekyc** [comment]
LGTM

**Pull request merged**